app/models/instance.rb

```diff
@@ -15,6 +15,9 @@ def find_tenant_by_host(host)
     end
   end
 
+  VALID_HOST_NAME_REGEX = /\A[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\z/i
+  validates :host, format: { with: VALID_HOST_NAME_REGEX }, length: { maximum: 255 }
```

Comment:
Can you provide any links ?

I still think trailing periods is invalid, and I also tried some validation
tools online to confirm.

On Thu, Jan 8, 2015 at 9:53 PM, Joel Low <notifications@github.com> wrote:

> In app/models/instance.rb
> <https://github.com/Coursemology/coursemology2/pull/41#discussion-diff-22652109>
> :
>
> > @@ -15,6 +15,9 @@ def find_tenant_by_host(host)
> >      end
> >    end
> >
> > +  VALID_HOST_NAME_REGEX = /\A[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\z/i
> > +  validates :host, format: { with: VALID_HOST_NAME_REGEX }, length: { maximum: 255 }
>
> What label do you call the root? I don't know. I'm guessing. But trailing
> periods make sense to me, I've seen them in use before so I've no reason to
> believe they're invalid. What you're looking for is probably normalisation,
> not validation.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Coursemology/coursemology2/pull/41/files#r22652109>.
>